feat(cms): new session strategy

nickhsine opened the pull request:
### 前情提要
此 PR 是為了實作少年報導者會員系統，可以參考[系統設計：少年報導者會員登入](https://www.notion.so/twreporter/25a8dbdca9328095a080ce900800cc43)了解全貌。

### PR 說明
實作目標是讓使用者可以用 go-api 核發的 access_token 來與 kids-cms Keystone GraphQL API 溝通。

現行的 kids-api 如果要與 kids-cms 溝通，預設的作法，是先透過 headless account 的 Admin 帳號密碼來換取 session token，再將後續的 request 帶上 `cookie: keystonejs-session=<session token>` header，來通過 Keystone server 的 access control 打 GQL。

此 PR 的作法更改 Keystone 處理 session 的機制：如果進來的 request 帶有 `keystonejs-session` cookie 的話，就依照舊有 的方法驗證 Admin 身份和權限；但若是進來的 request 不是帶 `keystonejs-session` cookie，而是帶 `Authorization: Bearer <access_token>` 的話，就改用 jwt 來驗證會員（Member）的身份。 我們便能透過 Member 和 Admin 的區別來設定不同的 access control。

### 實作細節
1. 新增 Member（會員） list：存放會員資料和該名會員在報導者的 `user_id`
2. 修改 keystone session strategy，從 `statelessSession`（只驗 cookie） 改成 `compositeSession`（驗 cookie 和 access_token）

當 request 是搭配 `Authorization: Bearer <access_token>` 進到系統時，keystone 會先驗證 access_token 是否合格，如果合格，則解開 access_token 的 payload，查看 `user_id` claim，得到該 request 在報導者的 user id，接著查看少報是否已經為此 user 建立會員，如果已經建立，則用該會員的資料產生 keystone session；若沒有會員資料，則建立資料後，再產生 keystone session。後續的 resolvers 和 hooks 就能夠透過 keystone session 拿到該名會員的資訊。

go-api 實作 JWT access_token 時，採用的演算法是 HS256，若 kids-cms 如果要驗證 token，會需要  JWT secret，所以設計讓 kids-cms 需要從環境變數讀取 JWT secret。

註：
比較好的作法，是 go-api 在產生 JWT 時採用非對稱簽名的方式，如此一來，kids-cms 就不需要知道 JWT secret（私鑰），僅用公鑰就可以驗證 JWT。此優化可以排入後續任務。

nickhsine committed:
feat(cms): add member list and adopt composite session stragedy

- authenticate requests by `Authorization: Bearer <access_token>` header or keystone session cookie
- provision member if needed

nickhsine committed:
chore(cms): update schema.(graphql,prisma)

nickhsine committed:
chore(cms): add migrations/20250814091101_add_member_list/migration.sql

nickhsine committed:
chore(cms): update schema.(graphql,prisma)

nickhsine committed:
chore(cms): add migrations/20250826023350_update_member_list/migration.sql

nickhsine committed:
refactor(cms): add essential comments

howar31 commented on packages/cms/keystone.ts:
```diff
@@ -23,9 +26,173 @@ const { withAuth } = createAuth({
+      'DatabaseError',
+      'Error to find member or provision a member',
+      {
+        twreporter_user_id: decoded.twreporter_user_id,
```
nitpick:
`decoded` 內有 `twreporter_user_id` 嗎？還是這邊要記錄 `user_id` 呢？

nickhsine replied:
`decoded` 裡面沒有 `twreporter_user_id`，這邊要修改成 `decoded.user_id`。

howar31 reviewed (approved):
LGTM, 1 nitpick

nickhsine committed:
fix(keystone): restrict access to sensitive membership_user fields

- Added `access.read` restrictions on `twreporter_user_id` and `role` fields to only allow Admin/Owner roles.
- Disabled `create` and `update` operations on these fields to prevent changes.
